### ISSUE TITLE
SecretsManager List Secrets Filter with Key all deviated from boto3

### ISSUE THREAD

**Issue by obsidian33** [getmoto/moto#8421]
## How to reproduce

```python
@pytest.fixture
def secretsmanager_client():
  with mock_aws():
    yield boto3.client("secretsmanager", region_name="us-east-1")
    
def test_list_secrets(secretsmanager_client):
  secretsmanager_client.create_secret(Name="ThisTestSecret")
  secretsmanager_client.create_secret(Name="RealSecret")
  secretsmanager_client.create_secret(Name="NexttestSecret")
  
  response = secretsmanager_client.list_secrets(
    Filters=[{"Key": "all", "Values": ["test"]}]
  )
```

With the Filter using the `Key` of `all` boto3 documentation states:
> Breaks the filter value string into words and then searches all attributes for matches. Not case-sensitive.

The response from moto returns zero results (`'SecretList': []`) when it should return two secret objects with the Names: `ThisTestSecret` and `NexttestSecret`. 

I've tested this against an AWS environment and boto3 works as described and does return the expected results.

## Versions

Installed using poetry:
- moto v5.0.24
- boto3 v1.35.83

**Comment by bblommers** [comment]
Hi @obsidian33, thanks for raising this. 

> Breaks the filter value string into words

It took me some time to figure this out, but I think that AWS determines words by splitting values that are upper-cased. 

 - `ThisTestSecret` is split into three words, `This`, `Test` and `Secret`.
 - `Nexttestsecret` will be broken down into two words, `Nexttest` and `Secret`.

When I execute your example against AWS, it only returns `ThisTestSecret`. I'm assuming that's because of a typo, and you meant `NextTestSecret`? That splits into `Next`, `Test` and `Secret`, so that would be found by AWS.

Moto currently doesn't do the word break correctly, we only break words that are separated by a space, so I'll mark this as a bug.

**Comment by obsidian33** [comment]
@bblommers yea sorry that was a typo but you've got it figured out!

I did find another edge case. I can open up a separate issue if you'd like but boto3 also splits words by other special characters that are supported in the secret name. 

According to AWS the secret name allows `/_+=.@-` characters. AWS recommends using slashes, e.g. `prod/AppBeta/Mysql`. I've confirmed that it also splits this in to `prod`, `AppBeta`, and `Mysql` but filtering for the `/` character returns zero results. I am not exactly sure about the other special characters though.

**Comment by bblommers** [comment]
@obsidian33 Good point, I didn't think about special characters. I've raised another PR to also split values on those characters - let me know if anything is still wrong/missing (after the next release).
